Pocket/pocket-ios
# feat: cards display in iPad

## Summary
Update card display for iPad view for both our slate detail (see all) sections and collection stories section. 

Currently the pages have a single column layout for cards, which is okay for smaller screens, but we want to improve the experience when viewing these pages on big surfaces.

## References 
IN-1577

## Implementation Details
Modified the section layout code for `CollectionViewController` and `SlateDetailViewController`. The code between the two are fairly similar besides the fact that they take in two different core data models, `CollectionStory` and `Recommendation` respectively. 

Steps to determine what layout to show:
1. Check whether device is using wide layout (based on device being iPad and has horizontal size class of regular)
2. Display either `sectionForWideLayout` or `sectionForCompact` depending on requirement in 1. Note: `sectionForCompact` was mainly a copy and paste of existing code into its own function.
3. If we want a wide layout that has a 3 column grid layout, there are 2 requirements: the view's width length must be large enough and the device must be in landscape view; otherwise, the wide layout uses a 2 column grid layout.
4. We create the section based on list of items and height of the cells. For the height, we need to calculate the row heights by retrieving the max height of the items in a row using `getMaxHeightForRow`.

Update: After some conversations, refactored the code to live in a class called `GridSectionLayout`. Logic remains the same, but organizing code and renaming differed.

## Test Steps
Navigate to Home and tap on "See All" for the different sections and view the recommendations layout. Verify the cards display experience in various devices and modes (i.e. iphone mini, iPhone plus, iPad Pro, iPad mini, Split View, landscape, portrait). 

Navigate to a Collection whether in Home, Saves or Search and view the stories layout. Verify the cards display experience in various devices and modes (i.e. iphone mini, iPhone plus, iPad Pro, iPad mini, Split View, landscape, portrait). 

## PR Checklist:

## Screenshots
| iPad Landscape | iPad Portrait |
| --- | --- |
| ![image](https://github.com/Pocket/pocket-ios/assets/6743397/bc692815-446a-4536-9283-b73b1d2280f3) | ![image](https://github.com/Pocket/pocket-ios/assets/6743397/23a51aa5-6bf5-461b-82ee-0aae9fb6933b) |
